src/pybind/mgr/ansible_orchestrator/module.py

```diff
@@ -0,0 +1,364 @@
+"""
+ceph-mgr Ansible orchestrator module
+
+The external Orchestrator is the Ansible runner service (RESTful https service)
+"""
+
+# Python stuff
+from threading import Event
+import errno
+import json
+
+# Ceph stuff
+from mgr_module import MgrModule
+import orchestrator
+
+# Orchestrator stuff
+# A Client is used to communicate with the Ansible Runner service
+from ansible_runner_svc import Client, PlayBookExecution, ExecutionStatusCode,\
+                               EVENT_DATA_URL
+
+# Constants section
+
+# Time to clean the completions list
+WAIT_PERIOD = 10
+
+
+# List of playbooks names used
+GET_INVENTORY_PLAYBOOK = "probe-disks.yml"
```

Comment:
can you share this file as an example?

Reply:
This is playbook i'm using in this moment  is only providing "free disks". In any case it was good enough to be used as base for implement the "get_inventory" method. I'm modifying it in order to get a list of all the devices. ( in any case if you know other playbook with this function it will be welcome)


```
[jolmomar@localhost tmp]$ cat probe-disks.yml
---
#
# Playbook to scan a set of hosts and return a dict indexed by host containing
# a list of disks that are unused. Each disk is represented by a dict with the
# following fields;
#
# size_txt (str) e.g 10.0GB
# size_bytes (int) e.g. 21474836480
# sectorsize (int) e.g. 512
# sectors (int) e.g 41943040
#
# example output;
# ok: [con-1 -> 127.0.0.1] => {
#    "free_disks": {
#        "con-1": {
#            "vdd": {
#                "rotational": true,
#                "sectors": 41943040,
#                "sectorsize": 512,
#                "size_bytes": 21474836480,
#                "size_txt": "20.00 GB"
#            }
#        },

- name: probe hosts for free disks
  hosts:
    - osds
    - mgrs
    - mons
  vars:
    free_disks: |
      {%- set disk_table = dict() %}
      {%- for host in play_hosts %}
        {%- set _x = disk_table.__setitem__(host, {}) %}
        {%- set _devdata = dict() %}
        {%- for disk in hostvars[host].host_disk %}
            {%- set _meta = hostvars[host]['ansible_devices'][disk] %}
            {%- set _x = _devdata.__setitem__(disk, dict(size_txt=_meta['size'],
                                                         rotational=_meta['rotational']|bool,
                                                         sectors=_meta['sectors']|int,
                                                         sectorsize=_meta['sectorsize']|int,
                                                         size_bytes=_meta['sectors']|int * _meta['sectorsize']|int)) %}
        {%- endfor %}
        {%- set _x = disk_table.__setitem__(host, _devdata) %}
      {%- endfor %}
      {{ disk_table }}

  gather_facts: true
  tasks:
    - name: setup
      set_fact:
          host_disk: []
    - name: Get a list of block devices (excludes loop and child devices)
      command: lsblk -n --o NAME --nodeps --exclude 7
      register: lsblk_out
    - name: check if disk {{ item }} is free
      command: pvcreate --test /dev/{{ item }}
      ignore_errors: true
      register: pv_status
      with_items: "{{lsblk_out.stdout_lines}}"
    - name: Update hosts freedisk list
      set_fact:
        host_disk: "{{host_disk + [item.item]}}"
      ignore_errors: true
      when: item.rc == 0
      with_items: "{{ pv_status.results}}"
    - name: RESULTS
      debug:
        var: free_disks
      delegate_to: 127.0.0.1
      run_once: True
```

Reply:
Changed the name of the playbook to "host-disks.yml"